src/skeleton/generate_data.cc

```diff
@@ -473,6 +473,7 @@ Ret emit_data(Skeleton& skel) {
     FILE* keys = fopen (keys_name.c_str(), "wb");
     if (!keys) {
         RET_FAIL(error("cannot open file: %s", keys_name.c_str()));
+        fclose(input);
```

Comment:
`fclose` needs to go before `RET_FAIL`, as the macro returns.

Reply:
Haha, sorry for that.
Fixed.